### ISSUE TITLE
RSS <ttl> Element handling

### ISSUE THREAD

**Issue by vatonbero** [newsboat/newsboat#2074]
Hi there,

I am using Newsboat 2.27.0. In my config i have the following:
```
reload-time 720
```
While I enjoy this option in configuration, several websites of mine are using time to live RSS element eg. `<ttl>$SOME_VALUE</ttl>`. 

This got me thinking: Since I found no keywords in the docs nor source repo, is newsboat honouring time to live in RSS feeds? If not, can this be added to the program with honouring it by default, with option to override it in program config?

Here is one reference of this RSS element with example: https://www.w3schools.com/XML/rss_tag_ttl.asp

Am I not aware of any Atom equivalents so nothing to do there.

**Comment by vatonbero** [comment]
@Minoru if you ask me personally, I would expect for newsboat to compare ttl and `reload-time` values and not reload feed if `reload-time` value is lower than ttl. 

Example:
1. `reload-time` value is 7min and ttl value is 10min
2. at 7th minute newsboat would reload all feeds without ttl (10min)
3. at 14th minute newsboat would reload all feeds with one using ttl

**Comment by Minoru** [comment]
Yeah, I am asking you personally, because you're the author of the issue and you're the most knowledgeable about what exactly you envision :)

Your example makes sense. I guess Newsboat should also take `reload` and `reload-all` operations into account, i.e. the TTL timer should start after the last fetch regardless of the way in which that fetch was initiated. I also think we'll also need a way to force the reload, for cases when the user *really* wants to know if the feed was updated.

Newsboat already implements HTTP conditional requests, which save a ton of traffic if there are no updates, so I don't think this issue is too pressing. But I'll merge a PR that implements this nonetheless!

**Comment by bogdasar1985** [comment]
@Minoru If processing of `ttl` gonna be added it should be ignored by default, but forced by user via `config` or `url` files. Because if we suddenly turn on their processing, users will not expect this.

**Comment by Minoru** [comment]
You have a point, but OTOH users won't bother looking for a setting unless they absolutely need it. I think it should be on by default, but:

1. TTL should be capped to some reasonable value, like one day. This helps with misconfigured feeds that have very large values (e.g. if the admin thought that the value is measured in seconds);
2. there should be a way to reload a particular feed regardless of its TTL. Perhaps `reload` (<kbd>r</kbd>) should do that, while `reload-all` (<kbd>^R</kbd>) should honour the TTL;
3. there should be a way to ignore TTL for certain feeds that are known to have misconfigured TTLs, e.g. if a "firehose" feed from something like Hacker News had a TTL of a few hours. This is needed so users don't have to reload this feed manually all the time. I'm not sure if it should be implemented as a setting. We could instead point people to `filter:`, and encourage them to write scripts that modify the TTL.

Now that I spell it all out, I start to wonder if it's even worth it. HTTP caching solves a very similar problem, and does it better because no-one has to predict how often the feed is going to be updated.

**Comment by Minoru** [comment]
Out of curiosity, I checked how many feeds in my urls file specify a TTL, and what value they use:

```
$ wc -l ~/.newsboat/urls
666 /home/minoru/.newsboat/urls
$ cat ~/.newsboat/urls | cut -f1 -d' ' | tr '\n' '\0' | xargs -0 curl --silent | grep --ignore-case --only-matching '<ttl>[^<]*</ttl>' | sort | uniq -c | sort --human-numeric-sort
      1 <ttl>120</ttl>
      1 <ttl>1440</ttl>
      1 <ttl>180</ttl>
      1 <ttl>600</ttl>
      7 <ttl>60</ttl>
     18 <ttl>15</ttl>
     24 <ttl>40</ttl>
```

So 8% of my feeds have it, and the majority of them specify 1 hour or less.

**Comment by Seirdy** [comment]
Another related type of markup is the [RSS Syndication module](https://web.resource.org/rss/1.0/modules/syndication/), available as an XML namespace.

A live example is available in the Atom and RSS feeds of seirdy.one.
